File: Python/test_responder.py
from request_response import Request, RequestContext
from responder import WikiPageResponder, SearchResponder
from wiki import WikiPage


def test_wiki_page_responder_ok():
    responder = WikiPageResponder()
    request = Request(request_type="GET", uri="/")
    root_page = WikiPage(title="FrontPage", uri="/")
    context = RequestContext(root_page)

    response = responder.make_response(request, context)

    assert response.http_code == "200"


def test_wiki_page_responder_ko():
    responder = WikiPageResponder()
    request = Request(request_type="GET", uri="/not-found")
    root_page = WikiPage(title="FrontPage", uri="/")
    context = RequestContext(root_page)

    response = responder.make_response(request, context)

    assert response.http_code == "404"
    assert response.page.title == "404"


def test_search_responder_no_results():
    responder = SearchResponder()
    request = Request(request_type="POST", uri="/", data={"search_text": "term"})
    root_page = WikiPage(title="FrontPage", uri="/")
    context = RequestContext(root_page)

    response = responder.make_response(request, context)

    assert response.page.text == "found term in pages:<ul></ul>"


def test_search_responder_with_results():
    responder = SearchResponder()
    request = Request(request_type="POST", uri="/", data={"search_text": "spam"})
    root_page = WikiPage(title="SpamPage", uri="/spam", text="spam")
    context = RequestContext(root_page)

    response = responder.make_response(request, context)

    assert response.page.text == "found term in pages:<ul><li>SpamPage</li></ul>"

File: Java/src/test/java/wikisearch/TestResponder.java
package wikisearch;

import org.junit.Test;

import java.util.HashMap;
import java.util.HashSet;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class TestResponder {

    @Test
    public void testWikiPageResponderOk() {
        Responder responder = new WikiPageResponder();
        Request request = new Request("GET", "/");
        WikiPage rootPage = new WikiPage("FrontPage", "", new HashSet<>(), "/");
        RequestContext context = new RequestContext(rootPage);

        Response response = responder.makeResponse(request, context);

        assertThat(response.httpCode, is("200"));
    }

    @Test
    public void testWikiPageResponderKo() {
        Responder responder = new WikiPageResponder();
        Request request = new Request("GET", "/not-found");
        WikiPage rootPage = new WikiPage("FrontPage", "", new HashSet<>(), "/");
        RequestContext context = new RequestContext(rootPage);

        Response response = responder.makeResponse(request, context);

        assertThat(response.httpCode, is("404"));
        assertThat(response.page.getTitle(), is("404"));
    }

    @Test
    public void testSearchResponderNoResults() {
        Responder responder = new SearchResponder();
        Request request = new Request("POST", "/", new HashMap<String, String>() {{
            put("search_text", "term");
        }});
        WikiPage rootPage = new WikiPage("FrontPage", "", new HashSet<>(), "/");
        RequestContext context = new RequestContext(rootPage);

        Response response = responder.makeResponse(request, context);

        assertThat(response.page.getText(), is("found term in pages:<ul></ul>"));
    }

    @Test
    public void testSearchResponderWithResults() {
        Responder responder = new SearchResponder();
        Request request = new Request("POST", "/", new HashMap<String, String>() {{
            put("search_text", "spam");
        }});
        WikiPage rootPage = new WikiPage("SpamPage", "spam", new HashSet<>(), "/spam");
        RequestContext context = new RequestContext(rootPage);

        Response response = responder.makeResponse(request, context);

        assertThat(response.page.getText(), is("found term in pages:<ul><li>SpamPage</li></ul>"));
    }
}
